src/zenml/integrations/modal/step_operators/modal_step_operator.py

```diff
@@ -0,0 +1,217 @@
+                    step_name=step_name,
+                )
+                builds.append(build)
+
+        return builds
+
+    def launch(
+        self,
+        info: "StepRunInfo",
+        entrypoint_command: List[str],
+        environment: Dict[str, str],
+    ) -> None:
+        """Launch a step run on Modal.
+
+        Args:
+            info: The step run information.
+            entrypoint_command: The entrypoint command for the step.
+            environment: The environment variables for the step.
+
+        Raises:
+            subprocess.CalledProcessError: If the modal app exits with a non-zero code.
+            ValueError: If no Docker credentials are found for the container registry.
+        """
+        settings = cast(ModalStepOperatorSettings, self.get_settings(info))
+        image_name = info.get_image(key=MODAL_STEP_OPERATOR_DOCKER_IMAGE_KEY)
+        zc = Client()
+        stack = zc.active_stack
+
+        if not stack.container_registry:
+            raise ValueError(
+                "No Container registry found in the stack. "
+                "Please add a container registry and ensure "
+                "it is correctly configured."
+            )
+
+        if docker_creds := stack.container_registry.credentials:
+            docker_username, docker_password = docker_creds
+        else:
+            raise RuntimeError(
+                "No Docker credentials found for the container registry."
+            )
+
+        my_secret = modal.secret._Secret.from_dict(
+            {
+                "REGISTRY_USERNAME": docker_username,
+                "REGISTRY_PASSWORD": docker_password,
+            }
+        )
+
+        spec = modal.image.DockerfileSpec(
+            commands=[f"FROM {image_name}"], context_files={}
+        )
+
+        zenml_image = (
+            modal.Image._from_args(
+                dockerfile_function=lambda *_, **__: spec,
+                force_build=False,
+                image_registry_config=modal.image._ImageRegistryConfig(
+                    api_pb2.REGISTRY_AUTH_TYPE_STATIC_CREDS, my_secret
+                ),
+            )
+            .env(environment)
+            .run_commands(" ".join(entrypoint_command))
+        )
+
+        resource_settings = info.config.resource_settings
+        gpu_str = settings.gpu
+        if resource_settings.gpu_count is not None:
+            gpu_str += f":{resource_settings.gpu_count}"
+
+        app = modal.App(f"zenml-{info.run_name}-{info.step_run_id}")
+
+        async def run_sandbox():
+            with modal.enable_output():
+                async with app.run():
+                    await modal.Sandbox.create.aio(
+                        "bash",
+                        "-c",
+                        " ".join(entrypoint_command),
+                        image=zenml_image,
+                        gpu=gpu_str,
+                        cpu=resource_settings.cpu_count or None,
```

Comment:
the `or None` is not necessary I think, `cpu_count` will already be `None` if it's not set.